framework/src/utils/MooseUtils.C

```diff
@@ -809,6 +809,22 @@ template std::string join<std::vector<MooseEnumItem>>(const std::vector<MooseEnu
 template std::string join<std::set<MooseEnumItem>>(const std::set<MooseEnumItem> &,
                                                    const std::string &);
 
+bool
+file_exists(const std::string & filename)
+{
+  struct stat buffer;
+  return (stat(filename.c_str(), &buffer) == 0);
```

Comment:
We've found that this is insufficient since the existence of a file, doesn't mean you can read or write to it. Use the existing method instead.

Reply:
OK